larray_editor/arraywidget.py

```diff
@@ -779,87 +780,96 @@ def set_data(self, data=None, bg_value=None):
                 else:
                     filters_layout.addWidget(QLabel("too big to be filtered"))
             filters_layout.addStretch()
-        self.data_adapter.update_filtered_data({})
+
+        self.gradient_chooser.setEnabled(self.model_data.bgcolor_possible)
+
+        self.data_adapter.set_data(data, bg_value=bg_value)
 
         # reset default size
         self.view_axes.set_default_size()
         self.view_ylabels.set_default_size()
         self.view_xlabels.set_default_size()
         self.view_data.set_default_size()
 
-    def _update_digits_scientific(self, data):
+    # called by set_data and ArrayEditorWidget.accept_changes (this should not be the case IMO)
+    # two cases:
+    # * set_data should update both scientific and ndigits
+    # * toggling scientific checkbox should update only ndigits
```

Comment:
So you want to split `_update_digits_scientific` in 2 methods?

Reply:
Ideally yes, but this is tricky to do efficiently and elegantly because there is some computation that needs to be shared for updating scientific and ndigits. It used to be split in more methods but I just remerged them for efficiency reasons. It should probably be possible to have code that is both clean and efficient but I have not spent the time to reach that point and found the efficient version "decently clean".